base/shared/src/main/scala/scalaz/BaseHierarchy.scala

```diff
@@ -12,6 +12,10 @@ object BaseHierarchy {
     implicit def monadApply[M[_]](implicit M: Monad[M]): Apply[M] = M.applicative.apply
     implicit def monadFunctor[M[_]](implicit M: Monad[M]): Functor[M] = M.applicative.apply.functor
     implicit def monoidSemigroup[A](implicit A: Monoid[A]): Semigroup[A] = A.semigroup
+    implicit def groupMonoid[A](implicit A:Group[A]):Monoid[A] = A.monoid
+    implicit def abelianGroupGroup[A](implicit A:AbelianGroup[A]):Group[A] = A.group
+    implicit def ringAbelianGroup[A](implicit A:Ring[A]):AbelianGroup[A] = A.abelianGroup
+    implicit def ringMonoid[A](implicit A:Ring[A]):Monoid[A] = A.monoid
```

Comment:
This worries me; a `Ring` contains two distinct `Monoid`s, and this picks the "multiplicative" one. However, `groupMonoid compose abelianGroupGroup compose ringAbelianGroup` _also_ produces a `Monoid` from a `Ring`.  This will cause the somewhat counter-intuitive behavior where asking for `Monoid[A]` gives you the multiplicative one, but calling a method that asks for `Group[A]` and then internally uses `Monoid[A]` will find the additive one.

I think we should have a rule such as

```haskell
newtype Mult a = Mult a
instance Ring a => Monoid (Mult a) where
  ...
```

using the (as yet forthcoming) tag types.